python/composio/tools/local/sqltool/actions/sql_query.py

```diff
@@ -25,37 +26,52 @@ class SqlQueryResponse(BaseModel):
 
 class SqlQuery(LocalAction[SqlQueryRequest, SqlQueryResponse]):
     """
-    Executes a SQL Query and returns the results
+    Executes a SQL Query and returns the results for both local SQLite and remote databases
     """
 
     _tags = ["sql", "sql_query"]
 
-    def execute(self, request: SqlQueryRequest, metadata: Dict) -> SqlQueryResponse:
-        """Execute SQL query"""
-        # Implement logic to process input and return output
-        # Example:
-        # response_data = {"result": "Processed text: " + request.text}
-        # Implement logic to process input and return output
-
-        # Check if the database file exists
-        if not Path(request.connection_string).exists():
-            raise ValueError(
-                f"Error: Database file '{request.connection_string}' does not exist."
-            )
+    def _is_sqlite_connection(self, connection_string: str) -> bool:
+        """Determine if the connection string is for a SQLite database"""
+        return (
+            connection_string.endswith(".db")
+            or connection_string.endswith(".sqlite")
+            or connection_string.endswith(".sqlite3")
+            or connection_string.startswith("sqlite:///")
+        )
 
+    def execute(self, request: SqlQueryRequest, metadata: Dict) -> SqlQueryResponse:
+        """Execute SQL query for either SQLite or remote databases"""
         try:
-            # Use 'with' statement to manage the database connection
-            with sqlite3.connect(request.connection_string) as connection:
-                cursor = connection.cursor()
-                cursor.execute(request.query)
-                response_data = cursor.fetchall()
-                connection.commit()
-
-            # Prepare the response data
-            return SqlQueryResponse(
-                execution_details={"executed": True},
-                response_data=response_data,
-            )
-
-        except sqlite3.Error as e:
-            raise ValueError(f"SQLite error: {str(e)}") from e
+            if self._is_sqlite_connection(request.connection_string):
+                return self._execute_sqlite(request)
+            else:
+                return self._execute_remote(request)
+        except Exception as e:
```

Comment:
Consider using `sqlalchemy.exc.SQLAlchemyError` for more specific error handling in remote database operations instead of catching generic Exception. This would provide better error messages and handling for database-specific issues.

Reply:
Added